### DISCUSSION TITLE
Remote cache miss problems — trying to debug

### DISCUSSION THREAD

**Discussion by robations** [vercel/turborepo#1607] [Help]
Hello hello, I went through the process of setting up Turborepo a short while back and set up remote caching (uisng Vercel). It seemed like local caching was working consistently and speeding up builds 🚀. I had to add and adjust my `.gitignores` to make sure global gitignores were accounted for, but with this done all seemed okay.

A few days after merging these changes I noticed that hardly any of the remote caches were being hit. I assumed it was due to stray config files or something like that on one server, but having had a couple of hours to dig into I just seem to have different cache keys for no apparent reason.

Let's say I've got two servers that are running builds for the same git hash. I get a different hash key for `packages/foo` for the `build` task.

What I've tried:

- `git status` — no extraneous files on either server
- enumerating all files and hashing:
  - `git ls-files | sort | xargs sha1sum | sha1sum` (for example)
  - after a slight issue with sorting (solved by `LC_ALL=C`) I got the same sha1 on both servers
- No environment variables are dependent for the task
- Taken a quick look at the [source code here](https://github.com/vercel/turborepo/blob/main/cli/internal/taskhash/taskhash.go) but I only have a vague knowledge of Go at the moment.
- The only file differences I can see are in `packages/foo/node_modules/.bin/` where the NODE_PATH is specific to the server. But `node_modules` is git-ignored in any case.

Both servers are running Ubuntu, different versions but wouldn't be expecting different dependencies to be installed because of this.

Any suggestions for debugging this further?

**Comment by tknickman** [comment]
Hey @robations, thanks for the report - what version of `turbo` are you using? Also, what type of apps (Next.js, svelte, etc.) are in your monorepo?

**Reply by robations** [reply]
Hi @tknickman thanks for the reply. I saw the issue with 1.3.1 and then upgraded to 1.4.0 with the same result. I see there are a couple of bug release versions published since last week, although [not marked in github yet](https://github.com/vercel/turborepo/issues/1606) (polite hint).

When I've finished my client work I'll try again with 1.4.2 (especially if you're aware of any fixes that might be related).

I suppose this could be a feature request to debug what goes into the hash. As soon as there are hash discrepancies, it's very had to pin this down to a file or directory.

> what type of apps

expressjs, angularjs (old school), custom webpack build, cli scripts, random nodejs typescript

**Reply by nathanhammond** [reply]
Where are you deploying to?￼ If Vercel, have you turned on analytics?

**Reply by nathanhammond** [reply]
(If you're able to reproduce with 1.3.1 then all of the automated environment variable things are not relevant.)

**Reply by nathanhammond** [reply]
Are the servers running the same git version?

**Comment by anthonyshew** [comment]
Closing due to no response.
